src/agentlab/agents/step_agent/step_agent.py

```diff
@@ -0,0 +1,109 @@
+import re
```

Comment:
[nit] Maybe you plan to do this already, but could you please include a header mentioning "Adapted from ..." or "Copied from ..." for all the relevant files.